gtfs/feed_source.py

```diff
@@ -26,10 +38,127 @@ def url(self) -> str:
     def bbox(self) -> Bbox:
         pass
 
-    def fetch(self):
-        """
-        Modify this method in subclass for importing feed(s) from agency.
+    def write_status(self):
+        """Write pickled log of feed statuses and last times files were downloaded."""
+        LOG.debug(f"Downloading finished. Writing status file {self.status_file}...")
+        with open(self.status_file, "wb") as status_file:
+            pickle.dump(self.status, status_file)
+            LOG.debug(f"Statuses written to {self.status_file}.")
 
-        By default, loops over given URLs, checks the last-modified header to see if a new
+    def fetch(self) -> bool:
+        """Modify this method in subclass for importing feed(s) from agency.
+
+        By default, checks the last-modified header to see if a new
         download is available, streams the download if so, and verifies the new GTFS.
         """
+        if self.url:
+            feed_file = self.__class__.__name__
+            if self.download_feed(feed_file, self.url):
+                self.write_status()
+                # if self.verify(feed_file):
+                #     LOG.info('GTFS verification succeeded.')
+                #     return True
+                # else:
+                #     LOG.error('GTFS verification failed.')
+                #     return False
+            else:
+                return False
+        else:
+            raise ValueError("URL not set for feed source!")
+
+    def check_header_newer(self, feed_file: str, url: str):
+        """Check if last-modified header indicates a new download is available.
+
+        :param feed_file: Name of downloaded file (relative to :ddir:)
+        :param url: Where GTFS is downloaded from
+        :returns: 1 if newer GTFS available; 0 if info missing; -1 if already have most recent
+        """
+        if not os.path.exists(self.status_file):
+            LOG.debug(f"Status file {self.status_file} not found.")
+            return 0
+
+        with open(self.status_file, "rb") as f:
+            last_status = pickle.load(f)
+        if feed_file in last_status and "posted_date" in last_status[feed_file]:
```

Comment:
what else is supposed to be in that status file?

Reply:
according to their repo there were other props like is_valid, is_effective and all but it was handled during the verification call  so right now it's just the posted date (last modified)

Reply:
jep ok thx, let's keep it for now, I think eventually we'll need smth to check if the date ranges in the gtfs dataset are valid right now